saleor/order/utils.py

```diff
@@ -576,15 +574,30 @@ def create_order_discount_for_order(
 ):
     """Add new order discount and update the prices."""
 
-    new_total = apply_discount_to_value(value, value_type, order.currency, order.total)
-    amount = (order.total - new_total).gross
+    current_total = order.total
+    currency = order.currency
+    if value_type == DiscountValueType.PERCENTAGE:
+        # percentage discount will generate different value for net and gross
+        net_total = apply_discount_to_value(
+            value, value_type, currency, current_total.net
+        )
+        gross_total = apply_discount_to_value(
+            value, value_type, currency, current_total.gross
+        )
+    else:
+        new_total = apply_discount_to_value(value, value_type, currency, current_total)
+        net_total = new_total.net
+        gross_total = new_total.gross
+
+    new_amount = (current_total - gross_total).gross
```

Comment:
The same block is used below in `update_order_discount_for_order`, maybe it should be a function?